core/src/odbc_uri.rs

```diff
@@ -73,7 +73,7 @@ impl ODBCUri {
         }
         let mut ret = ODBCUri::process_uri(odbc_uri.clone())?;
         if ret.get(DSN).is_some() {
-            let mut dsn_opts = DSNOpts::from_attribute_string(&odbc_uri);
+            let mut dsn_opts = DSN::from_attribute_string(&odbc_uri);
```

Comment:
hmm, interesting that rust can actually distinguish the DSN str from the DSN struct. The rust style guide would prefer Dsn for the struct (opposite of the go style guide): https://rust-lang.github.io/api-guidelines/naming.html

So lets go ahead and rename the struct DSN to Dsn.